lib/mongoid/factory.rb

```diff
@@ -63,12 +79,30 @@ def build(klass, attributes = nil)
     #
     # @return [ Document ] The instantiated document.
     def from_db(klass, attributes = nil, criteria = nil, selected_fields = nil)
+      execute_from_db(klass, attributes, criteria, selected_fields, execute_callbacks: true)
+    end
+
+    # Execute from_db.
+    #
+    # @param [ Class ] klass The class to instantiate from if _type is not present.
+    # @param [ Hash ] attributes The document attributes.
+    # @param [ Criteria ] criteria Optional criteria object.
+    # @param [ Hash ] selected_fields Fields which were retrieved via
+    #   #only. If selected_fields are specified, fields not listed in it
+    #   will not be accessible in the returned document.
+    # @param [ true | false ] execute_callbacks Flag specifies whether callbacks
+    #   should be run.
```

Comment:
To clarify the callback behavior, how about something like:

> Whether this method should invoke the callbacks. If true, the callbacks will be invoked normally. If false, the callbacks will be stored in the +pending_callbacks+ list and caller is responsible for invoking +run_pending_callbacks+ at a later time. Use this option to defer callback execution until the entire object graph containing embedded associations is constructed.

That's a lot of text to duplicate in all docstrings, perhaps one copy of it will suffice somewhere with appropriate references.

Reply:
I don't like doing references in the code in case that code is changed. I put the description in the code once and left everything else as is. Let me know if that's good enough or what kind of reference I can put everywhere else.

Reply:
That is good with me!